contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -353,28 +293,38 @@ contract IpRoyaltyVault is IIpRoyaltyVault, ERC20SnapshotUpgradeable, Reentrancy
         return _getIpRoyaltyVaultStorage().isClaimedAtSnapshot[snapshotId][claimer][token];
     }
 
-    /// @notice The list of revenue tokens in the vault
+    /// @notice Returns list of revenue tokens in the vault
     function tokens() external view returns (address[] memory) {
-        return _getIpRoyaltyVaultStorage().tokens.values();
+        return (_getIpRoyaltyVaultStorage().tokens).values();
     }
 
-    function _hasAncestorIp(address ipId, address ancestorIpId) internal returns (bool) {
-        (bool success, bytes memory returnData) = IP_GRAPH_CONTRACT.call(
-            abi.encodeWithSignature("hasAncestorIp(address,address)", ipId, ancestorIpId)
-        );
-        if (!success) revert Errors.IpRoyaltyVault__IpGraphCallFailed();
-        return abi.decode(returnData, (bool));
+    /// @notice A function to calculate the amount of revenue token claimable by a token holder at certain snapshot
+    /// @param account The address of the token holder
+    /// @param snapshotId The snapshot id
+    /// @param token The revenue token to claim
+    /// @return The amount of revenue token claimable
+    function _claimableRevenue(address account, uint256 snapshotId, address token) internal view returns (uint256) {
+        IpRoyaltyVaultStorage storage $ = _getIpRoyaltyVaultStorage();
+
+        // if the ip is tagged, then the unclaimed royalties are lost
+        if (DISPUTE_MODULE.isIpTagged($.ipId)) return 0;
+
+        uint256 balance = balanceOfAt(account, snapshotId);
+        uint256 totalSupply = totalSupplyAt(snapshotId);
+        uint256 claimableToken = $.claimableAtSnapshot[snapshotId][token];
+        return $.isClaimedAtSnapshot[snapshotId][account][token] ? 0 : (balance * claimableToken) / totalSupply;
```

Comment:
Could `totalySupply` be zero in extreme case?

Reply:
No because all the vaults deployed are initialized with `TOTAL_RT_SUPPLY`:

`IIpRoyaltyVault(ipRoyaltyVault).initialize("Royalty Token", "RT", TOTAL_RT_SUPPLY, ipId, receiver);`